aws_lambda_powertools/utilities/idempotency/persistence/base.py

```diff
@@ -183,7 +183,7 @@ def _get_hashed_idempotency_key(self, lambda_event: Dict[str, Any]) -> str:
 
     @staticmethod
     def is_missing_idempotency_key(data) -> bool:
-        return data is None or not data or all(x is None for x in data)
+        return data is None or not data or (type(data).__name__ in ("tuple", "list") and all(x is None for x in data))
```

Comment:
Can we break this into a multi-line thing?

It's quite complex now and we might find another bug here in the future I'm sure